app/src/controllers/reservation/services/ReservationService.ts

```diff
@@ -247,39 +247,92 @@ const ReservationService: ReservationServiceInterface = {
   },
 
   async updateReservation(user, shopId, reservationId, reservationDate, clientId, menuId, stylistId) {
-    if (user.role.slug === RoleSlug.SHOP_STAFF && !await isUserOwnedShop(user.id, shopId)) {
-      Logger.debug('Shop is not owned by user')
-      throw new AuthorizationError()
```

Comment:
this should still be used to check if the user owns this shop or not, maybe you can change the logic a bit